AURN_Data_Download/AURN_download.py

```diff
@@ -626,53 +665,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
     # If a single year is passed then convert to a list with a single value
     if type(years) is int:
         years = [years]
+    print('Years:', years)
     current_year = datetime.datetime.now()
 
 
     base_path = Path("AURN_data_download")
-    if (base_path.is_dir() is False): 
-        base_path.mkdir() 
-    # change this if we need more organised data storage later  
+    if (base_path.is_dir() is False):
+        base_path.mkdir()
+    # change this if we need more organised data storage later
     data_path = base_path
 
-    # get list of sites to process
-    site_list = metadata['AURN_metadata']['site_id'].unique()
-    
+    # get list of sites to processextract_site_data
+    if not site_list:
+        site_list = metadata['AURN_metadata']['site_id'].unique()
+    else:
+        # Todo - Test that sites provided are correct
+        pass
+    print('Site list', site_list)
+    #sys.exit(0)
+
     # create the station location dataset
     stations = metadata['AURN_metadata'][['site_id','latitude', 'longitude','site_name']].drop_duplicates()
     stations = stations.rename(columns={"site_id":"SiteID","latitude":"Latitude","longitude":"Longitude"})
     stations = stations.set_index('SiteID')
-    
-    
+
+
     # create a dataframe with the hourly dataset for all stations
-    #daily_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe= hourly_dataframe.rename(columns={'siteID':'SiteID'})
-    
+    #daily_dataframe = (site_list,metadata,years,data_path,save_to_csv)
```

Comment:
We should delete this commented out code completely - it's just me being untidy.

Reply:
Have removed that comment.